opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,385 @@
+                            % type(view_aggregation))
+
+        display_name_prefix = DEFAULT_DISPLAY_NAME_PREFIX
+        if self.options.metric_prefix != "":
+            display_name_prefix = self.options.metric_prefix
+
+        descriptor_pattern = "projects/%s/metricDescriptors/%s"
+        project_id = self.options.project_id
+
+        desc_labels = new_label_descriptors(self.default_labels, view.columns)
+
+        descriptor = monitoring_v3.types.MetricDescriptor(labels=desc_labels)
+        descriptor.type = metric_type
+        descriptor.metric_kind = metric_kind
+        descriptor.value_type = value_type
+        descriptor.description = view.description
+        descriptor.unit = unit
+
+        descriptor.name = descriptor_pattern % (project_id, metric_type)
+        descriptor.display_name = "%s/%s" % (display_name_prefix, view_name)
+
+        client = self.client
+        project_name = client.project_path(project_id)
+        descriptor = client.create_metric_descriptor(project_name, descriptor)
+        return descriptor
+
+
+def new_stats_exporter(options):
+    """ new_stats_exporter returns an exporter that
+        uploads stats data to Stackdriver Monitoring.
+    """
+    if str(options.project_id).strip() == "":
+        raise Exception(ERROR_BLANK_PROJECT_ID)
+
+    client = monitoring_v3.MetricServiceClient()
+
+    exporter = StackdriverStatsExporter(client=client, options=options)
+
+    if options.default_monitoring_labels is not None:
+        exporter.set_default_labels(options.default_monitoring_labels)
+    else:
+        label = {}
+        key = remove_non_alphanumeric(get_task_value())
+        label[key] = OPENCENSUS_TASK_DESCRIPTION
+        exporter.set_default_labels(label)
+    return exporter
+
+
+def get_task_value():
+    """ getTaskValue returns a task label value in the format of
+     "py-<pid>@<hostname>".
+    """
+    task_value = "py@" + str(os.getpid())
+    hostname = platform.uname()[1]
+    task_value += hostname if hostname is not None else "localhost"
+    return task_value
+
+
+def namespaced_view_name(view_name):
+    """ create string to be used as metric type
+    """
+    return os.path.join("custom.googleapis.com", "opencensus", view_name)
+
+
+def new_label_descriptors(defaults, keys):
+    """ create labels for the metric_descriptor
+        that will be sent to Stackdriver Monitoring
+    """
+    label_descriptors = []
+    for key, lbl in defaults.items():
+        label = {}
+        label["key"] = remove_non_alphanumeric(key)
+        label["description"] = lbl
+        label_descriptors.append(label)
+
+    for tag_key in keys:
+        label = {}
+        label["key"] = remove_non_alphanumeric(tag_key)
+        label_descriptors.append(label)
+    return label_descriptors
+
+
+def remove_non_alphanumeric(text):
+    """ Remove characters not accepted in labels key
+    """
+    return str(re.sub('[^0-9a-zA-Z ]+', '', text)).replace(" ", "")
+
+
+def is_float(value):
```

Comment:
This is a weird helper whose usage is quite awkward but also the body and the return arguments. `is_float` implies just checking if a value is a float not conversion as well. Let's also change the order of arguments returned

I think we want to make this something like
```python
def as_float(value):
     """ Converts a value to a float if possible
          On success, it returns (converted_value, True)
          On failure, it returns (None, False)
     """
     try:
          return float(value), True
      except: # Catch all exception including ValueError
          return None, False
```

and then we can use it like this
```python
convFloat, isFloat = as_float(tag_value[0])
if isFloat:
   point.value.double_value = convFloat
```

instead of
```python
 if is_float(tag_value[0])[0]:
    point.value.double_value = float(tag_value[0])
```
which unfortunately again runs the conversion but also for which the if condition isn't readable.

Reply:
Because of the nox lint, I was not allowed to use just except:, but instead I created using except Exception as exception: